tests/logs/push/test_ami_push/test_create_image_failure.txt

```diff
@@ -15,8 +16,10 @@
 [   DEBUG] {"arch": "x86_64", "image_id": "ami-1234567", "image_name": "ami-rhel", "product_name": "RHEL_HOURLY", "region": "region-1", "variant": "BaseOS", "version": "8.5.0"}
 [   ERROR] Failed to create image ami-1234567 in rhsm with error code 500
 [   ERROR] 
-[ WARNING] 500 Server Error: None for url: https://example.com/v1/internal/cloud_access_providers/amazon/amis
```

Comment:
I feel like there's been a regression in the logging here.  We don't get a message per retry about the failed request any more.  I know if it keeps failing, we eventually get an exception raised at the end with all the details, but that could potentially be a long time after the original error occurred; and if the retries succeed, it seems we won't get any messages regarding the failed requests.

I wouldn't call it mandatory, but it'd be nice to improve this.  I see the old code in push_to_region had something like:

```python
try:
    ... # do the upload
except [...] as exc:
    LOG.warning(str(exc))
    ... # logic relating to retry
```

Since you changed the way retry is handled you dropped the try/except completely.  But perhaps it should be kept for logging purposes, i.e. log the exception at warning level as it was before, and then re-raise (allowing the caller to decide whether or not to retry).

The way the exception was logged was also not really idiomatic as there is an exc_info parameter which can be passed to the logger methods to cause them to log the current exception, so fixing that as well, the resulting code might look something like:

```python
try:
    ... # do the upload
except:
    # log whatever's the current exception and then re-raise
    LOG.warning("Upload failed", exc_info=True)
    raise
```

Reply:
Thanks, I have added logging for exceptions.